presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1467,6 +1470,16 @@ public Expression rewriteIdentifier(Identifier reference, Void context, Expressi
         {
             List<Set<Expression>> computedGroupingSets = ImmutableList.of(); // empty list = no aggregations
 
+            boolean isGroupingOperationPresent = outputExpressions.stream()
+                    .anyMatch(GroupingOperationAnalyzer::containsGroupingOperation);
+
+            if (isGroupingOperationPresent && !node.getGroupBy().isPresent()) {
+                throw new SemanticException(
+                        INVALID_PROCEDURE_ARGUMENTS,
```

Comment:
That's what I originally had but @raghavsethi told me to use the above error code. After thinking about it for a while I agree with him. Introducing an error that will be used just for this particular case is too specific. See https://github.com/prestodb/presto/pull/5465#r70118205 and https://github.com/prestodb/presto/pull/5465#r70118573.